print-ascii.c

```diff
@@ -56,6 +63,69 @@
 #define HEXDUMP_HEXSTUFF_PER_LINE \
 		(HEXDUMP_HEXSTUFF_PER_SHORT * HEXDUMP_SHORTS_PER_LINE)
 
+
+/*
+ * The blow is_utf8_printable is taken from ngrep
+ *
+ * Check if a UTF-8 character sequence is printable using standard library functions.
+ * Returns the number of bytes in the UTF-8 character if printable, 0 otherwise.
+ * Also returns the display width (1 or 2 columns) via the width_out parameter.
+ *
+ * This uses mbrtowc() to convert multi-byte UTF-8 to wide char, then iswprint()
+ * to check if it's printable, and wcwidth() to get the display width.
+ */
+static u_int is_utf8_printable(const unsigned char *s, size_t max_len, int *width_out) {
+	if (!s || max_len == 0) return 0;
+
+	mbstate_t state = {0};
+	wchar_t wc;
+
+	size_t len = mbrtowc(&wc, (const char *)s, max_len, &state);
+
+	/* Check for errors and incomplete sequences */
+	if (len == (size_t)-1) {
+		/* Encoding error */
+		return 0;
+	}
+
+	if (len == (size_t)-2) {
+		/* Incomplete multi-byte sequence (need more bytes) */
+		return 0;
+	}
+
+	if (len == 0) {
+		/* Null character */
+		return 0;
+	}
+
+	/* Check if the wide character is printable */
+#if defined(_WIN32) || defined(_WIN64)
```

Comment:
Are there any compilers for Windows that define `_WIN64`, but don't define `_WIN32`, on "modern" Windows? ("modern" here means "not 16-bit Windows".)